### DISCUSSION TITLE
Is it possible to set different Application.Styles based on current Theme? Similar to ResourceDictionary.ThemeDictionaries.

### DISCUSSION THREAD

**Discussion by cirrusone** [AvaloniaUI/Avalonia#14395] [Q&A]
In `App.axaml` I can change the SimpleTheme brushes based on currently selected Theme using ResourceDictionary.ThemeDictionaries similar to the following...

```
<Application.Resources>
    <ResourceDictionary>
        <ResourceDictionary.ThemeDictionaries>

            <ResourceDictionary x:Key="Dark">
                <SolidColorBrush x:Key="ThemeAccentBrush" Color="Blue" />
            </ResourceDictionary>

            <ResourceDictionary x:Key="Light">
                <SolidColorBrush x:Key="ThemeAccentBrush2" Color="Red" />
            </ResourceDictionary>

            <ResourceDictionary x:Key="{x:Static themes:CustomThemes.Light2}">
                <SolidColorBrush x:Key="ThemeAccentBrush3" Color="Green"/>
            </ResourceDictionary>

        </ResourceDictionary.ThemeDictionaries>
    </ResourceDictionary>
</Application.Resources>
```

Is it possible to do something similar with styles in `<Application.Styles>`? Eg want to change the style setter to different brushes based on selected Theme, eg how would I do the following?

```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
	     xmlns:themes="clr-namespace:ProjectA.Themes"
             x:Class="ProjectA.App"
             RequestedThemeVariant="Dark">
	<Application.Resources>
		<ResourceDictionary>
			<ResourceDictionary.ThemeDictionaries>

				<ResourceDictionary x:Key="Dark">
					<SolidColorBrush x:Key="ThemeAccentBrush" Color="Blue" />
				</ResourceDictionary>

				<ResourceDictionary x:Key="Light">
					<SolidColorBrush x:Key="ThemeAccentBrush2" Color="Red" />
				</ResourceDictionary>

				<ResourceDictionary x:Key="{x:Static themes:CustomThemes.Light2}">
					<SolidColorBrush x:Key="ThemeAccentBrush3" Color="Green"/>
				</ResourceDictionary>

			</ResourceDictionary.ThemeDictionaries>
		</ResourceDictionary>
	</Application.Resources>


	<Application.Styles>

		<SimpleTheme />

		<!-- Is it possible to set Style based on current theme? -->
		<!-- Dark Style -->
		<Style Selector="Button">
			<Setter Property="Background" Value="{DynamicResource ThemeAccentBrush}"/>
		</Style>

		<!-- Light Style -->
		<Style Selector="Button">
			<Setter Property="Background" Value="{DynamicResource ThemeAccentBrush2}"/>
		</Style>

		<!-- Light2 Style -->
		<Style Selector="Button">
			<Setter Property="Background" Value="{DynamicResource ThemeAccentBrush3}"/>
		</Style>

	</Application.Styles>
</Application>
```

**Comment by maxkatz6** [comment]
> Is it possible to do something similar with styles in <Application.Styles>? Eg want to change the style setter to different brushes based on selected Theme, eg how would I do the following?

You would define different brushes in the resources. 

Alternatively, you can use property selector: `Selector="Button[(ThemeVariant.ActualThemeVariant)=Light]"`.

**Reply by cirrusone** [reply]
_You would define different brushes in the resources._ - I want to actually change the brush in the style and not just define the color of the brushes, `{DynamicResource ThemeAccentBrush} `-> `{DynamicResource ThemeAccentBrush3}`.  Is that possible? I have some them clashes when using external controls and would prefer to change style brush globally.

_Alternatively, you can use property selector: Selector="Button[ActualThemeVariant=Light]"._ - This doesn't work for me in `App.axaml`, I get the following error when I do this

Error:
`Error   ActualThemeVariant of Avalonia.Base:Avalonia.StyledElement doesn't seem to be an AvaloniaProperty Line 149, position 4.`

```
<Application.Styles>

	<SimpleTheme />

	<Style Selector="Button[ActualThemeVariant=Light]">
		<Setter Property="Background" Value="#fd7d00"/>
	</Style>

</Application.Styles>
```

**Reply by marc2911** [reply]
Any Update on this? Is there really no way to conditionally set a Style according to the ActualThemeVariant??
